udata/core/dataset/rdf.py

```diff
@@ -468,6 +468,62 @@ def title_from_rdf(rdf, url):
             return i18n._("Nameless resource")
 
 
+def access_rights_from_rdf(resource: RdfResource) -> set[str]:
+    """
+    Extract the access rights from a RdfResource
+    Cardinality is 0..n (although it should be 0..1 per the spec).
+    """
+    return rdf_unique_values(resource, DCT.accessRights, parse_label=True)
+
+
+def licenses_from_rdf(resource: RdfResource) -> set[str]:
+    """
+    Extract licences from a RDF distribution.
+    See `test_dataset_rdf.py > test_licenses_from_rdf` for examples of supported formats.
+    Cardinality is 0..n (although it should be 0..1 per the spec).
+    """
+    return rdf_unique_values(resource, DCT.license, parse_label=True)
+
+
+def rights_from_rdf(resource: RdfResource) -> set[str]:
+    """
+    Extract rights from a RDF distribution.
+    Cardinality is 0..n.
+    """
+    return rdf_unique_values(resource, DCT.rights, parse_label=True)
+
+
+def provenances_from_rdf(resource: RdfResource) -> set[str]:
+    """
+    Extract provenance from a RDF distribution.
+    Cardinality is 0..n.
+    """
+    return rdf_unique_values(resource, DCT.provenance, parse_label=True)
+
+
+def infer_dataset_access_rights(resources_access_rights: list[set]) -> set | None:
+    """
+    Infer the dataset access rights from a list of resources access rights.
+    If all resources have the same set of access rights return it.
+    """
+    if not resources_access_rights:
+        return
+    if resources_access_rights[0] == set.intersection(*resources_access_rights):
```

Comment:
```suggestion
    if set.union(*resources_access_rights) == set.intersection(*resources_access_rights):
```

My bad. I mistakenly put `resources_access_rights[0] == set.intersection(*resources_access_rights)` in a short-lived comment but seems I was too slow to fix it :/

Counter-example: `[{1,2}, {1,2,3}]`